samcli/lib/deploy/deployer.py

```diff
@@ -481,15 +499,20 @@ def wait_for_execute(self, stack_name: str, stack_operation: str, disable_rollba
             waiter.wait(StackName=stack_name, WaiterConfig=waiter_config)
         except botocore.exceptions.WaiterError as ex:
             LOG.debug("Execute stack waiter exception", exc_info=ex)
-            if disable_rollback:
+            if disable_rollback and on_failure is not FailureMode.DELETE:
                 msg = self._gen_deploy_failed_with_rollback_disabled_msg(stack_name)
                 LOG.info(self._colored.red(msg))
 
             raise deploy_exceptions.DeployFailedError(stack_name=stack_name, msg=str(ex))
 
-        outputs = self.get_stack_outputs(stack_name=stack_name, echo=False)
-        if outputs:
-            self._display_stack_outputs(outputs)
+        try:
+            outputs = self.get_stack_outputs(stack_name=stack_name, echo=False)
+            if outputs:
+                self._display_stack_outputs(outputs)
+        except DeployStackOutPutFailedError as ex:
+            # Show exception if we aren't deleting stacks
+            if not stack_operation == "DELETE":
```

Comment:
Change this to check the FailureMode enum instead of checking for "DELETE" since that is not being used.